docs/en/stack/data-frames/dataframe-examples.asciidoc

```diff
@@ -0,0 +1,362 @@
+First, we filter the source data such that excludes all the cancelled flights by 
+using a query filter, then transform the data to contain the distinct number of 
+flights, the sum of delayed minutes, and the sum of the flight minutes by air 
+carrier. Finally, we use a `bucket_script` to determine what percentage of the 
+flight time was actually delay.
+
+[source,js]
+----------------------------------
+POST _data_frame/transforms/_preview
+{
+  "source": {
+    "index": "kibana_sample_data_flights",
+    "query": { <1>
+      "bool": {
+        "filter": [
+          { "term":  { "Cancelled": false } }
+        ]
+      }
+    }
+  },
+  "dest" : {
+    "index" : "sample_flight_delays_by_carrier"
+  },
+  "pivot": {
+    "group_by": { <2>
+      "carrier": { "terms": { "field": "Carrier" }}
+    },
+    "aggregations": {
+      "flights_count": { "value_count": { "field": "FlightNum" }},
+      "delay_mins_total": { "sum": { "field": "FlightDelayMin" }},
+      "flight_mins_total": { "sum": { "field": "FlightTimeMin" }},
+      "delay_time_percentage": { <3>
+        "bucket_script": {
+          "buckets_path": {
+            "delay_time": "delay_mins_total.value",
+            "flight_time": "flight_mins_total.value"
+          },
+          "script": "(params.delay_time / params.flight_time) * 100"
+        }
+      }
+    }
+  }
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+<1> Filter the source data to select only flights that were not cancelled.
+<2> The data is grouped by the `Carrier` field which contains the airline name.
+<3> This `bucket_script` performs calculations on the results that returned by 
+the aggregation, in this particular example to calculate what percentage of 
+travel time was taken up by delays.
+
+The API returns the following response:
+
+[source,js]
+----------------------------------
+{
+  "preview" : [
+    {
+      "carrier" : "ES-Air",
+      "flights_count" : 2802.0,
+      "flight_mins_total" : 1436927.5130677223,
+      "delay_time_percentage" : 9.335543983955839,
+      "delay_mins_total" : 134145.0
+    },
+    {
+      "carrier" : "JetBeats",
+      "flights_count" : 2833.0,
+      "flight_mins_total" : 1451143.6898144484,
+      "delay_time_percentage" : 8.937088787987832,
+      "delay_mins_total" : 129690.0
+    },
+    {
+      "carrier" : "Kibana Airlines",
+      "flights_count" : 2832.0,
+      "flight_mins_total" : 1419081.404241085,
+      "delay_time_percentage" : 9.088273556017194,
+      "delay_mins_total" : 128970.0
+    },
+    {
+      "carrier" : "Logstash Airways",
+      "flights_count" : 2914.0,
+      "flight_mins_total" : 1503620.8713908195,
+      "delay_time_percentage" : 9.544959286661593,
+      "delay_mins_total" : 143520.0
+    }
+  ]
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+The example above transforms flight data into a entity centric index for 
```

Comment:
Unless this first sentence would be meaningful to data scientists, I'd omit it since in my opinion it's unnecessary.